featuregate/gates.go

```diff
@@ -19,14 +19,54 @@ import (
 	"sync"
 )
 
+// Stage represents the Gate's lifecycle and what is the expected state of it.
+type Stage int8
+
+const (
+	// Alpha is used when creating a new feature and the Gate must be explicitly enabled
+	// by the operator.
+	//
+	// The Gate will be disabled by default.
+	Alpha Stage = iota
+	// Beta is used when the feature flag is well tested and is enabled by default,
+	// but can be disabled by a Gate.
+	//
+	// The Gate will be enabled by default.
+	Beta
+	// Stable is used when feature is permanently enabled and can not be disabled by a Gate.
+	// This value is used to provide feedback to the operator that the gate will be removed in the next version.
+	//
+	// The Gate will be enabled by default and will return an error if modified.
+	Stable
+)
+
+var stageNames = map[Stage]string{
+	Alpha:  "Alpha",
+	Beta:   "Beta",
+	Stable: "Stable",
+}
+
+func (s Stage) String() string {
+	if v, ok := stageNames[s]; ok {
+		return v
+	}
+	return "unknown"
+}
+
 // Gate represents an individual feature that may be enabled or disabled based
 // on the lifecycle state of the feature and CLI flags specified by the user.
 type Gate struct {
 	ID          string
 	Description string
-	Enabled     bool
+	// Deprecated: [v0.64.0] Use Stage instead, this field will move to be non exported.
+	Enabled        bool
+	Stage          Stage
+	IssueLinks     []string
+	RemovalVersion string
 }
 
+func (g Gate) IsEnabled() bool { return g.Enabled }
```

Comment:
I'm marking the field deprecated since the flag stage defines if the gate starts as either enabled or disabled by default.

Setting enabled would be counter productive at that point.

Reply:
What about implementing `IsEnabled()` by looking at the `Stage`?

Reply:
Not exactly sure how that would work. 

I should say `Enabled` will still be use, it just wont be publicly accessible by the developer